ethereum_history_api/circuits/lib/src/rlp/types.nr

```diff
@@ -25,26 +25,61 @@ struct RlpFragment {
 }
 
 impl RlpFragment {
-    fn assert_eq_bytes<FIELD_NAME_LEN, MAX_RLP_LEN, MAX_VALUE_LEN>(
+    fn assert_eq_bytes_length<FIELD_NAME_LEN, MAX_RLP_LEN, MAX_VALUE_LEN>(
         self,
         field_name: str<FIELD_NAME_LEN>,
         rlp: [u8; MAX_RLP_LEN],
-        value: BoundedVec<u8, MAX_VALUE_LEN>
+        value: [u8; MAX_VALUE_LEN],
+        length: u64
     ) {
         assert(self.data_type == STRING, f"{field_name}: Invalid RLP type");
-        assert(self.length == value.len, f"{field_name}: Invalid RLP length");
+        assert(self.length == length, f"{field_name}: Invalid RLP length");
         assert(
-            sub_array_equals_up_to_length(value.storage, rlp, self.offset, self.length), f"{field_name}: Invalid RLP value"
+            sub_array_equals_up_to_length(value, rlp, self.offset, self.length), f"{field_name}: Invalid RLP value"
         );
     }
 
-    fn assert_eq_string<FIELD_NAME_LEN, MAX_RLP_LEN, MAX_VALUE_LEN>(
+    fn assert_eq_bounded_vec<FIELD_NAME_LEN, MAX_RLP_LEN, MAX_VALUE_LEN>(
         self,
         field_name: str<FIELD_NAME_LEN>,
         rlp: [u8; MAX_RLP_LEN],
         value: BoundedVec<u8, MAX_VALUE_LEN>
     ) {
-        self.assert_eq_bytes(field_name, rlp, value);
+        self.assert_eq_bytes_length(field_name, rlp, value.storage, value.len);
+    }
+
+    fn assert_eq_null<FIELD_NAME_LEN, MAX_RLP_LEN>(self, field_name: str<FIELD_NAME_LEN>) {
+        assert(self.data_type == STRING, f"{field_name}: Invalid RLP type");
+        assert(self.length == 0, f"{field_name}: Invalid RLP length");
+    }
+
+    fn assert_eq_u1<FIELD_NAME_LEN, MAX_RLP_LEN>(
+        self,
+        field_name: str<FIELD_NAME_LEN>,
+        rlp: [u8; MAX_RLP_LEN],
+        value: u1
+    ) {
+        assert(self.data_type == STRING, f"{field_name}: Invalid RLP type");
+        assert(self.length == 1, f"{field_name}: Invalid RLP length");
+        assert_eq(value, (rlp[0] % 2) as u1, f"{field_name}: Invalid RLP value");
```

Comment:
If `value == 1` and `rlp == [3]` this function will pass assertions although it shouldn't. I think we have to add `assert(rlp[0] < 2)`

Reply:
It was overcomplicated. I will simplify it.